### DISCUSSION TITLE
Dynamic Navbar in Layout, getting pathname server side

### DISCUSSION THREAD

**Discussion by majolo** [vercel/next.js#56478] [App Router]
I want to add a Navbar to my application, and since it renders above all pages it seems logical to include the navbar in the layout. However I want the current page to be highlighted on the navbar and getting the pathname doesn't seem possible server side at the moment.

Does anyone have a good solution to this?

**Comment by aelew** [comment]
You can create a Navbar client component that uses usePathname() and import that into your layout.

**Reply by dhmacs** [reply]
What if you need to fetch some data to display the title (server side)?
